craft_providers/lxd/remotes.py

```diff
@@ -155,14 +157,19 @@ def add_remote(self, lxc: LXC) -> None:
 }
 
 
-def get_remote_image(provider_base: str) -> RemoteImage:
+def get_remote_image(provider_base: Union[Base, str]) -> RemoteImage:
     """Get a RemoteImage for a particular provider base.
 
     :param provider_base: string containing the provider base
 
     :returns: the RemoteImage for the provider base
     """
-    image = _PROVIDER_BASE_TO_LXD_REMOTE_IMAGE.get(provider_base)
+    # temporary backward compatibility before 2.0
+    if isinstance(provider_base, str):
```

Comment:
Same as above - what do you think about a `logger.warning()` deprecation notice here too?